config/config.go

```diff
@@ -1,31 +1,23 @@
 package config
 
-import (
-	"time"
-)
-
-// Configuration defines the standard config for this operator.
-type Configuration struct {
-	Report               bool
-	StoragePath          string
-	Interval             time.Duration
-	Endpoint             string
-	ReportEndpoint       string
-	ReportPullingDelay   time.Duration
-	ReportMinRetryTime   time.Duration
-	ReportPullingTimeout time.Duration
-	Impersonate          string
+// Configurator An interface for handling the upload configuration
+type Configurator interface {
+	IsEnabled() bool
```

Comment:
I am thinking whether we need this or not. I guess I would leave this config for the caller of the lib.  WDYT?

Reply:
Left it in as a simple way to test usage of the library to trigger the reportToLogs dry-run. But will also be useful for enable/disable when adding back the `Run()` flow.

Reply:
OK :-)